src/commands/getrange.rs

```diff
@@ -0,0 +1,204 @@
+use bytes::Bytes;
+use std::sync::{Arc, Mutex};
+use std::usize;
+
+use crate::commands::executable::Executable;
+use crate::commands::{CommandParser, CommandParserError};
+use crate::frame::Frame;
+use crate::store::Store;
+use crate::Error;
+
+/// Returns the substring of the string value stored at key, determined by the offsets start and
+/// end (both are inclusive). Negative offsets can be used in order to provide an offset starting
+/// from the end of the string. So -1 means the last character, -2 the penultimate and so forth.
+/// The function handles out of range requests by limiting the resulting range to the actual length
+/// of the string.
+///
+/// Ref: <https://redis.io/docs/latest/commands/getrange/>
+#[derive(Debug, PartialEq)]
+pub struct Getrange {
+    pub key: String,
+    pub start: i64,
+    pub end: i64,
+}
+
+impl Executable for Getrange {
+    fn exec(self, store: Arc<Mutex<Store>>) -> Result<Frame, Error> {
+        let store = store.lock().unwrap();
+        let value = store.get(&self.key);
+        let bytes = match value {
+            Some(val) => val,
+            None => return Ok(Frame::Bulk(Bytes::new())),
+        };
+
+        let value = String::from_utf8(bytes.to_vec()).unwrap();
+        // TODO: Should we worry about this conversion?
+        let len = value.len() as i64;
+
+        let start = get_positive_index(len, self.start);
+        let end = get_positive_index(len, self.end);
+
+        let subset: String = value
+            .chars()
+            // We don't care about out of range indexes, take and skip will handle it.
+            .take((end + 1) as usize)
+            .skip(start as usize)
+            .collect();
+
+        Ok(Frame::Bulk(Bytes::from(subset)))
+    }
+}
+
+fn get_positive_index(str_len: i64, index: i64) -> i64 {
+    let is_positive = index >= 0;
+    if is_positive {
+        index
+    } else {
+        str_len - index.abs()
+    }
+}
+
+impl TryFrom<&mut CommandParser> for Getrange {
+    type Error = Error;
+
+    fn try_from(parser: &mut CommandParser) -> Result<Self, Self::Error> {
+        let key = parser.next_string()?;
+        let start = parser.next_string()?;
+        let end = parser.next_string()?;
+
+        let start: i64 = start
+            .parse()
+            .map_err(|_| CommandParserError::InvalidCommandArgument { argument: start })?;
+
+        let end: i64 = end
+            .parse()
+            .map_err(|_| CommandParserError::InvalidCommandArgument { argument: end })?;
+
+        Ok(Self { key, start, end })
+    }
```

Comment:
Could we use `parser.next_integer` here?

Reply:
Of course, I was hoping to use that at some point, and when I got the chance, I didn't hehe.